source/common/network/dns_impl.cc

```diff
@@ -170,6 +175,8 @@ ActiveDnsQuery* DnsResolverImpl::resolve(const std::string& dns_name,
   } else {
     pending_resolution->getHostByName(AF_INET6);
   }
+  // Enable timer to wake us up if the request times out.
+  updateAresTimer();
```

Comment:
I think this should only be set in the `} else {` clause below, when we do asynchronous resolution. If we resolve immediately, there is no need for a timer.